src/mixins/java/org/spongepowered/common/mixin/core/network/FriendlyByteBufMixin.java

```diff
@@ -43,7 +49,87 @@ public abstract class FriendlyByteBufMixin implements FriendlyByteBufBridge {
 
     @ModifyVariable(method = "writeComponent", at = @At("HEAD"), argsOnly = true)
     private Component localizeComponent(final Component input) {
-        return NativeComponentRenderer.apply(input, this.impl$locale == null ? Locales.EN_US : this.impl$locale);
+        return NativeComponentRenderer.apply(input, this.impl$locale == null ? Locales.DEFAULT : this.impl$locale);
+    }
+
+    @Redirect(method = "writeItem", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/item/ItemStack;getTag()Lnet/minecraft/nbt/CompoundTag;"))
+    public CompoundTag renderItemComponents(final ItemStack stack) {
+        CompoundTag tag = stack.getTag();
+        if (tag != null && tag.contains(Constants.Item.ITEM_DISPLAY, 10)) {
```

Comment:
To remove branching, might be worth checking for the inverse and returning the tag right away.